src/app/pages/settings/settings.component.ts

```diff
@@ -1,14 +1,65 @@
 import { Component, OnInit } from '@angular/core';
+import { AuthService } from '../../core/auth.service';
+import { MsgIoService } from '../../core/msg-io.service';
+import { take } from 'rxjs/operators';
+import { DomSanitizer } from '@angular/platform-browser';
+
+declare var $: any;
 
 @Component({
   selector: 'app-settings',
   templateUrl: './settings.component.html',
   styleUrls: [ './settings.component.css' ]
 })
 export class SettingsComponent implements OnInit {
+  downloadMessagesData;
+  downloadUserData;
+  exportErrorText: string;
+
+  constructor(private authService: AuthService, private msgIoService: MsgIoService, private sanitizer: DomSanitizer) { }
 
-  constructor() { }
+  private static decodeHtml(html: string) {
+    const txt = document.createElement('textarea');
+    txt.innerHTML = html;
+    return txt.value;
+  }
 
   ngOnInit(): void { }
 
+  /**
+   * Converts the user's data and messages into JSONs and changes the href of their respective buttons to permit
+   * downloading of the data.
+   */
+  async generateDownloads() {
+    const downloadModal = $('#download');
+    const exportErrorModal = $('#download-error');
+    try {
+      // Get data
+      // Looser types is to allow the message array to be replaced with a single error if there are no messages
+      let messages: Array<any> | void = await this.msgIoService.getMessages().pipe(take(1)).toPromise();
+      const userData = await this.authService.getUserData();
+      // Convert encoded text to normal text
+      if (messages) {
+        messages.forEach(message => {
+          message.contents = SettingsComponent.decodeHtml(message.contents);
+        });
+      } else {
+        messages = [{
+          error: 'No messages found'
+        }];
```

Comment:
```suggestion
        messages = [{ error: 'No messages found' }];
```
Or is this against Code Reformat rules?